src/components/cc-env-var-editor-expert/cc-env-var-editor-expert.types.d.ts

```diff
@@ -0,0 +1,13 @@
+import { EnvVarMode, Variable } from "../common.types";
+
+type EnvVarEditorExpertState = EnvVarEditorExpertStateLoading | EnvVarEditorExpertStateLoaded;
+
+interface EnvVarEditorExpertStateLoading {
+  type: 'loading';
+}
+
+interface EnvVarEditorExpertStateLoaded {
+  type: 'loaded';
+  mode: EnvVarMode;
+  variables: Array<Variable>;
+}
```

Comment:
note: technically, these 3 types are the same for simple, expert and json (and it's by design so we can interchange them)

question: should we move these types to the common one and reuse it in the 3 components?

Reply:
I think this is a good idea

Reply:
I'll make separate commit for that